internal/core/workers/operation_execution_worker/operation_execution_worker_test.go

```diff
@@ -490,4 +522,95 @@ func TestSchedulerOperationsExecutionLoop(t *testing.T) {
+
+		config := mockconfig.NewMockConfig(mockCtrl)
+		config.EXPECT().GetDuration(gomock.Any()).Return(duration)
+
+		operationName := "test_operation"
+		operationDefinition := mockoperation.NewMockDefinition(mockCtrl)
+		operationExecutor := mockoperation.NewMockExecutor(mockCtrl)
+		operationManager := mock.NewMockOperationManager(mockCtrl)
+
+		defFunc := func() operations.Definition { return operationDefinition }
+		definitionConstructors := operations.NewDefinitionConstructors()
+		definitionConstructors[operationName] = defFunc
+
+		scheduler := &entities.Scheduler{Name: "random-scheduler"}
+
+		executors := map[string]operations.Executor{}
+		executors[operationName] = operationExecutor
+		workerService := NewOperationExecutionWorker(scheduler, workers.ProvideWorkerOptions(config, operationManager, executors, nil, nil))
+		pendingOpsChan := make(chan string)
+
+		operationManager.EXPECT().PendingOperationsChan(gomock.Any(), gomock.Any()).Return(pendingOpsChan)
+		operationManager.EXPECT().CreateOperation(gomock.Any(), gomock.Any(), gomock.Any()).Do(func(_ context.Context, schedulerName string, op operations.Definition) {
+			require.Equal(t, scheduler.Name, schedulerName)
+			require.IsType(t, &healthcontroller.SchedulerHealthControllerDefinition{}, op)
+		}).Return(&operation.Operation{}, nil).AnyTimes()
+
+		ctx, cancel := context.WithCancel(context.Background())
+
+		go func() {
+			time.Sleep(5 * time.Millisecond)
+			cancel()
+		}()
+
+		err = workerService.Start(ctx)
+		require.NoError(t, err)
+	})
+
+	t.Run("when healthcontroller ticks and CreateOperation return in error, continue the execution normally", func(t *testing.T) {
+		duration, err := time.ParseDuration("1ms")
+		require.NoError(t, err)
+
+		mockCtrl := gomock.NewController(t)
+
+		config := mockconfig.NewMockConfig(mockCtrl)
+		config.EXPECT().GetDuration(gomock.Any()).Return(duration)
+
+		operationName := "test_operation"
+		operationDefinition := mockoperation.NewMockDefinition(mockCtrl)
+		operationExecutor := mockoperation.NewMockExecutor(mockCtrl)
+		operationManager := mock.NewMockOperationManager(mockCtrl)
+
+		defFunc := func() operations.Definition { return operationDefinition }
+		definitionConstructors := operations.NewDefinitionConstructors()
+		definitionConstructors[operationName] = defFunc
+
+		scheduler := &entities.Scheduler{Name: "random-scheduler"}
+
+		executors := map[string]operations.Executor{}
+		executors[operationName] = operationExecutor
+		workerService := NewOperationExecutionWorker(scheduler, workers.ProvideWorkerOptions(config, operationManager, executors, nil, nil))
+		pendingOpsChan := make(chan string)
+
+		operationManager.EXPECT().PendingOperationsChan(gomock.Any(), gomock.Any()).Return(pendingOpsChan)
+		executionCount := 0
+		operationManager.EXPECT().CreateOperation(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(func(_ context.Context, schedulerName string, op operations.Definition) (*operation.Operation, error) {
```

Comment:
Again, no need to use Do() here, we can assert the types directly replacing de gomock.Anys() and just return an error in every call, the **executionCount** var is not being used here.